### ISSUE TITLE
a11y-light fails WCAG AA contrast with gray background

### ISSUE THREAD

**Issue by gabalafou** [Quansight-Labs/accessible-pygments#30]
Some of the text color choices fail the 4.5:1 contrast ratio required to meet WCAG AA when the background color is changed from white to gray. 

You can see this failure in a [code block on the PyData Sphinx Theme website](https://pydata-sphinx-theme.readthedocs.io/en/latest/examples/kitchen-sink/blocks.html#code-block) (make sure you are looking at the light theme version, not dark theme).

<img width="751" alt="screenshot of the code block" src="https://github.com/Quansight-Labs/accessible-pygments/assets/317883/b78ea3b7-8503-4576-926b-dd7d951c5e16">

In that example, the shade of blue used for class and function names, #007faa, against the background gray, #f3f4f5, produces a contrast of 4.1:1, which fails the contrast requirement for regular size text.

**Comment by gabalafou** [comment]
cc @trallard

**Comment by trallard** [comment]
Noting that I plan to work on this mid/late November

**Comment by pradyunsg** [comment]
👋🏽 Apologies for the drive-by comment.

> You can see this failure in a [code block on the PyData Sphinx Theme website](https://pydata-sphinx-theme.readthedocs.io/en/latest/examples/kitchen-sink/blocks.html#code-block) (make sure you are looking at the light theme version, not dark theme).

This seems to be due to pydata-sphinx-theme overriding the theme's background configuration by setting `background-color: var(--pst-color-surface);` on `<pre>` tags. You can see that the `.highlight` parent of the `<pre>` tag sets the background color according to the pygments styling.

That CSS variable is related to a user-configurable setting on pydata-sphinx-themes' end (https://pydata-sphinx-theme.readthedocs.io/en/latest/user_guide/styling.html#color-variables).

**Comment by trallard** [comment]
Hey @pradyunsg, lovely to see you around and thanks for sharing your insights!

> This seems to be due to pydata-sphinx-theme overriding the theme's background configuration by setting background-color: var(--pst-color-surface); on <pre> tags. 

We are aware of this, and while that could be easily fixed by changing this overwrite in PST, we also realised the themes could improve their contrasts. 
Other people/themes are likely customising the background colour, so the upcoming changes in #33  should help cover a broader range of background colours and providing good defaults for some of these customisation cases (note the "some" because we will never be able to support al flavours of customisations)
